src/extensions/ipc/ipc-main.ts

```diff
@@ -19,27 +19,43 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 import { ipcMain } from "electron";
-import { IpcPrefix, IpcStore } from "./ipc-store";
-import { Disposers } from "./lens-extension";
-import type { LensMainExtension } from "./lens-main-extension";
+import { IpcPrefix, IpcRegistrar } from "./ipc-registrar";
+import { Disposers } from "../lens-extension";
+import type { LensMainExtension } from "../lens-main-extension";
+import type { Disposer } from "../../common/utils";
+import { once } from "lodash";
 
-export abstract class MainIpcStore extends IpcStore {
+export abstract class IpcMain extends IpcRegistrar {
   constructor(extension: LensMainExtension) {
     super(extension);
-    extension[Disposers].push(() => MainIpcStore.resetInstance());
+    extension[Disposers].push(() => IpcMain.resetInstance());
   }
 
-  handleIpc(channel: string, handler: (event: Electron.IpcMainInvokeEvent, ...args: any[]) => any): void {
+  /**
+   * Listen for broadcasts within your extension
+   * @param channel The channel to listen for broadcasts on
+   * @param listener The function that will be called with the arguments of the broadcast
+   * @returns An optional disopser, Lens will cleanup when the extension is disabled or uninstalled even if this is not called
+   */
+  listenIpc(channel: string, listener: (event: Electron.IpcRendererEvent, ...args: any[]) => any): Disposer {
     const prefixedChannel = `extensions@${this[IpcPrefix]}:${channel}`;
+    const cleanup = once(() => ipcMain.removeListener(prefixedChannel, listener));
 
-    ipcMain.handle(prefixedChannel, handler);
-    this.extension[Disposers].push(() => ipcMain.removeHandler(prefixedChannel));
+    ipcMain.addListener(prefixedChannel, listener);
+    this.extension[Disposers].push(cleanup);
+
+    return cleanup;
   }
 
-  listenIpc(channel: string, listener: (event: Electron.IpcMainEvent, ...args: any[]) => any): void {
+  /**
+   * Declare a RPC over `channel`. Lens will cleanup when the extension is disabled or uninstalled
+   * @param channel The name of the RPC
+   * @param handler The remote procedure that is called
+   */
+  handleIpc(channel: string, handler: (event: Electron.IpcMainInvokeEvent, ...args: any[]) => any): void {
```

Comment:
Why aren't we returning a disposer in this case too?

Reply:
I can add some documentation here. But RPC is quite different than IPC. I think I will change the method names too.

Namely, because it is an error to call `invokeRpc` without an "alive" `handleRpc` it doesn't lend itself to being as transient as `listenIpc`. 

Furthermore, the use case surrounding early cleanup of `listenIpc` doesn't apply to `handleRpc` as `main` has no concept of components mounting and unmounting on the DOM.

Reply:
OK, makes sense. Some mention in the docs about this would good.